File: dotnet/cafdemalihapa/Directories/Hosting/Feature/NotesAndMessages/NotesAndMessagesDirectory.cs
using cafdemalihapa.Files;

namespace cafdemalihapa.Directories.Hosting.Feature.NotesAndMessages
{
    public class NotesAndMessagesDirectory(
            FeatureDirectory featureDirectory
        )
    {
        public string GetPath()
        {
            var directoryName = "notes and messages";
            var featureDirectoryPath = featureDirectory.GetPath();
            var notesAndMessages = Path.Combine(featureDirectoryPath, directoryName);
            return notesAndMessages;
        }
        public void Create()
        {
            var path = GetPath();
            if(!Directory.Exists(path))
            {
                Directory.CreateDirectory(path);
            }
        }

        public void CreateFiles()
        {
            var path = GetPath();
            FileService.CreateNumberedFiles(path);
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Hosting/Feature/NotesAndMessages/NotesAndMessagesDirectory.fs
namespace cafdemwimu.console.Directories.Hosting.Feature.NotesAndMessages

open System.IO
open cafdemwimu.console.Files
open cafdemwimu.console.Directories.Hosting.Feature

type NotesAndMessagesDirectory() =
    member _.GetPath() =
        let directoryName = "notes and messages"
        let featureDirectoryPath = FeatureDirectory.GetPath()
        let notesAndMessages = Path.Combine(featureDirectoryPath, directoryName)
        notesAndMessages

    member this.Create() =
        let path = this.GetPath()
        if not (Directory.Exists(path)) then
            Directory.CreateDirectory(path) |> ignore

    member this.CreateFiles() =
        let path = this.GetPath()
        FileService.CreateNumberedFiles path
